File: Climbing Stairs.java
public class Solution {
    public int climbStairs(int n) {
        // Start typing your Java solution below
        // DO NOT write main() function
        if(n==0)
        {
            return 0;
        }
        int []dp = new int[2];
        dp[0] = 1;
        dp[1] = 1;
        int ret = 1;
        for(int i = 1; i < n; i++)
        {
            ret = dp[0] + dp[1];
            dp[1] = dp[0];
            dp[0] = ret;
        }
        
        return ret;
    }
}

File: Climbing Stairs.cpp
class Solution {
public:
    int climbStairs(int n) {
        // Start typing your C/C++ solution below
        // DO NOT write int main() function
        if(n < 2)
            return 1;
        
        vector<int> dp(n+1, 0);
        dp[0] = 1;
        dp[1] = 1;
        
        for(int i = 2; i < n+1; i++)
        {
            dp[i] = dp[i-1]+dp[i-2];
        }
        return dp[n];
    }
};
